src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -122,19 +130,27 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 		{
 			if (ctx.useMaterials)
 			{
-				char name[WORD_BUFFER_LENGTH];
-				bufPtr = goAndCopyNextWord(name, bufPtr, WORD_BUFFER_LENGTH, bufEnd);
+				bufPtr = goAndCopyNextWord(tmpbuf, bufPtr, WORD_BUFFER_LENGTH, bufEnd);
 #ifdef _IRR_DEBUG_OBJ_LOADER_
-				os::Printer::log("Reading material _file",name);
+				os::Printer::log("Reading material _file",tmpbuf);
 #endif
-				readMTL(ctx, name, relPath);
+
+                SAssetLoadParams loadParams;
+                auto bundle = AssetManager->getAsset(tmpbuf, loadParams);
```

Comment:
shouldn't you be using the variant where you can specify the hierarchy level ? you're loading from a dependent loader

Also:
https://github.com/buildaworldnet/IrrlichtBAW/pull/410/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h#r362125278